contracts/finance/VestingWalletConfidentialFactory.sol

```diff
@@ -0,0 +1,175 @@
+    error VestingWalletConfidentialInvalidStartTimestamp(address beneficiary, uint64 startTimestamp);
+
+    /**
+     * @dev
+     */
+    event VestingWalletConfidentialBatchFunded(ebool success);
+    /**
+     * @dev
+     */
+    event VestingWalletConfidentialCreated(
+        address indexed beneficiary,
+        address indexed vestingWalletConfidential,
+        uint48 startTimestamp
+    );
+
+    /**
+     * @dev
+     */
+    struct VestingPlan {
+        address beneficiary;
+        externalEuint64 encryptedAmount;
+        uint48 startTimestamp;
+        uint48 cliff;
+        address executor;
+    }
+
+    /**
+     * @dev
+     */
+    constructor() {
+        _vestingWalletConfidentialImplementation = address(new VestingWalletCliffExecutorConfidential());
+    }
+
+    /**
+     * @dev Batches the funding of multiple confidential vesting wallets.
+     *
+     * Funds are sent to predeterministic wallet addresses. Wallets can be created later.
+     */
+    function batchFundVestingWalletConfidential(
+        address confidentialFungibleToken,
+        externalEuint64 totalEncryptedAmount,
+        bytes calldata inputProof,
+        VestingPlan[] calldata vestingPlans,
+        uint48 durationSeconds
+    ) external returns (ebool) {
+        require(durationSeconds > 0, VestingWalletConfidentialInvalidDuration());
+        euint64 totalTransferedAmount = euint64.wrap(0);
+        uint256 vestingPlansLength = vestingPlans.length;
+        for (uint256 i = 0; i < vestingPlansLength; i++) {
+            VestingPlan memory vestingPlan = vestingPlans[i];
+            euint64 encryptedAmount = FHE.fromExternal(vestingPlan.encryptedAmount, inputProof);
+            require(
+                vestingPlan.startTimestamp >= block.timestamp,
+                VestingWalletConfidentialInvalidStartTimestamp(vestingPlan.beneficiary, vestingPlan.startTimestamp)
+            );
+            address vestingWalletConfidential = predictVestingWalletConfidential(
+                vestingPlan.beneficiary,
+                vestingPlan.startTimestamp,
+                durationSeconds,
+                vestingPlan.cliff,
+                vestingPlan.executor
+            );
+            FHE.allow(encryptedAmount, confidentialFungibleToken);
+            euint64 transferredAmount = IConfidentialFungibleToken(confidentialFungibleToken).confidentialTransferFrom(
+                msg.sender,
+                vestingWalletConfidential,
+                encryptedAmount
+            );
+            totalTransferedAmount = FHE.select(
+                FHE.eq(encryptedAmount, transferredAmount),
+                FHE.add(totalTransferedAmount, transferredAmount),
+                FHE.asEuint64(0)
+            );
+        }
+        // Revert batch if one failed?
+        ebool success = FHE.eq(totalTransferedAmount, FHE.fromExternal(totalEncryptedAmount, inputProof));
+        emit VestingWalletConfidentialBatchFunded(success);
```

Comment:
We don't have the ability to revert based on confidential conditions. I think we should eliminate this check.